crates/matrix-sdk/src/notification_settings.rs

```diff
@@ -0,0 +1,1156 @@
+        }
+    }
+
+    #[async_test]
+    async fn delete_user_defined_room_rules() {
+        let server = MockServer::start().await;
+        let client = logged_in_client(Some(server.uri())).await;
+        Mock::given(method("PUT")).respond_with(ResponseTemplate::new(200)).mount(&server).await;
+        Mock::given(method("DELETE")).respond_with(ResponseTemplate::new(200)).mount(&server).await;
+
+        let mut ruleset = client.account().push_rules().await.unwrap();
+        let notification_settings = client.notification_settings();
+        let room_id = "!test_room:matrix.org".to_string();
+
+        let mode =
+            notification_settings.get_user_defined_room_notification_mode(&room_id, &ruleset);
+        assert!(mode.is_none());
+
+        // Add a Room rule
+        let result = notification_settings.insert_room_rule(&room_id, false, &mut ruleset).await;
+        assert!(result.is_ok());
+
+        // Add a an Override rule
+        let result =
+            notification_settings.insert_override_room_rule(&room_id, false, &mut ruleset).await;
+        assert!(result.is_ok());
+
+        let custom_rules = notification_settings.get_custom_rules_for_room(&room_id, &ruleset);
+        assert_eq!(custom_rules.len(), 2);
+
+        // Delete the custom rules
+        let result =
+            notification_settings.delete_user_defined_room_rules(&room_id, &mut ruleset).await;
+        assert!(result.is_ok());
+
+        let custom_rules = notification_settings.get_custom_rules_for_room(&room_id, &ruleset);
+        assert!(custom_rules.is_empty());
+    }
+
+    #[async_test]
+    async fn set_room_notification_mode_default_to_all_messages() {
+        let server = MockServer::start().await;
+        let client = logged_in_client(Some(server.uri())).await;
+
+        Mock::given(method("PUT")).respond_with(ResponseTemplate::new(200)).mount(&server).await;
+        Mock::given(method("DELETE")).respond_with(ResponseTemplate::new(200)).mount(&server).await;
+
+        let mut ruleset = client.account().push_rules().await.unwrap();
+        let notification_settings = client.notification_settings();
+        let room_id = "!test_room:matrix.org".to_string();
+
+        // Default -> All
+        let mode =
+            notification_settings.get_user_defined_room_notification_mode(&room_id, &ruleset);
+        assert!(mode.is_none());
+
+        let result = notification_settings
+            .set_room_notification_mode(&room_id, RoomNotificationMode::AllMessages, &mut ruleset)
+            .await;
+        assert!(result.is_ok());
+
+        let expected_mode = &RoomNotificationMode::AllMessages;
+        match notification_settings.get_user_defined_room_notification_mode(&room_id, &ruleset) {
+            Some(new_mode) => {
+                assert_eq!(
+                    &new_mode, expected_mode,
+                    "mode {:?} is expected (got {:?})",
+                    expected_mode, &new_mode
+                )
```

Comment:
`assert_eq` already prints the mismatching values if it fails, so I think this can just be

```suggestion
                assert_eq!(&new_mode, expected_mode);
```

Reply:
Ok, I've updated the tests accordingly.